source/docs/software/basic-programming/joystick.rst

```diff
@@ -0,0 +1,151 @@
+
+The USB Devices Tab contains indicators of the values of axes buttons and the POV that can be used to determine the mapping between physical joystick features and axis or button numbers. Simply click the joystick in the list to select it and the indicators will begin responding to the joystick input.
+
+``Joystick`` Class
+------------------
+
+.. image:: images/joystick/joystick.png
+
+.. tabs::
+
+   .. code-tab:: java
+
+      Joystick exampleJoystick = new Joystick(0); // 0 is the USB Port to be used as indicated on the Driver Station
+
+   .. code-tab:: c++
+
+      Joystick exampleJoystick{0}; // 0 is the USB Port to be used as indicated on the Driver Station
+
+The ``Joystick`` class is designed to make using a flight joystick to operate the robot significantly easier.  Depending on the flight joystick, the user may need to set the specific X, Y, Z, and Throttle channels that your flight joystick uses.  This class offers special methods for accessing the angle and magnitude of the flight joystick.
+
+``XboxController`` Class
+------------------------
+
+.. image:: images/joystick/xbox.jpg
+
+.. tabs::
+
+   .. code-tab:: java
+
+      XboxController exampleXbox = new XboxController(0); // 0 is the USB Port to be used as indicated on the Driver Station
+
+   .. code-tab:: c++
+
+      XboxController exampleXbox{0}; // 0 is the USB Port to be used as indicated on the Driver Station
+
+The ``XboxController`` class provides named indicies for each of the buttons that you can access with ``XboxController.Button.kX.value``.  The rumble feature of the controller can be controlled by using ``XboxController.setRumble(GenericHID.RumbleType.kRightRumble, value)``.  Many users do a split stick arcade drive that uses the left stick for just forwards / backwards and the right stick for left / right turning.
+
+POV
+---
+
+.. image:: images/joystick/dpadangles.png
+   :alt: The angles used by the code of the POV/D-pad with 0 at the top and continuing clockwise.
+
+On joysticks the POV is directional hat that can select one of 8 different angles or read -1 for unpressed.  The XboxController D-pad works the same as a POV.  Be careful when using a POV with exact angle requirements as it is hard for the user to ensure they select exactly the angle desired.
+
+``GenericHID`` Usage
+--------------------
+
+An axis can be used with ``.getRawAxis(0)`` (if not using either of the classes above) that returns the current value.  Zero in this example is the index of axis as found in the Drive Station mentioned above.
```

Comment:
This section is much too simple. getRawAxis works for either of the other two classes as well. And to actually use ``GenericHID`` you have to implement the abstract methods.

Reply:
I missed that GenericHID was abstract.  I think it is important to detail something here but doing a full extension might be a bit much for an example.  Maybe we just cut out the example?

Reply:
I think the example is fine, I would just put it under a different heading